src/Microsoft.Dnx.Tooling/Building/BuildManager.cs

```diff
@@ -296,6 +310,68 @@ private bool BuildInternal(string projectPath)
             return success;
         }
 
+        private void AddPackageFiles(string projectDirectory, IEnumerable<KeyValuePair<string, PackageFilesSource>> packageFiles, PackageBuilder packageBuilder, IList<DiagnosticMessage> diagnostics)
+        {
+            var rootDirectory = new DirectoryInfoWrapper(new DirectoryInfo(projectDirectory));
+
+            foreach(var match in CollectAdditionalFiles(rootDirectory, packageFiles, _currentProject.ProjectFilePath, diagnostics))
+            {
+                packageBuilder.Files.Add(match);
+            }
+        }
+
+        internal static IEnumerable<PhysicalPackageFile> CollectAdditionalFiles(DirectoryInfoBase rootDirectory, IEnumerable<KeyValuePair<string, PackageFilesSource>> projectFileGlobs, string projectFilePath, IList<DiagnosticMessage> diagnostics)
+        {
+            foreach(var pair in projectFileGlobs)
+            {
+                // Evaluate the globs on the right
+                var matcher = new Matcher();
+                matcher.AddIncludePatterns(pair.Value.Values);
+                var results = matcher.Execute(rootDirectory);
+                var files = results.Files.ToList();
+
+                // Check the arity of the left
+                if(pair.Key.EndsWith("/"))
+                {
+                    var dir = pair.Key.Substring(0, pair.Key.Length - 1).Replace('/', Path.DirectorySeparatorChar);
+
+                    foreach(var file in files)
+                    {
+                        yield return new PhysicalPackageFile()
+                        {
+                            SourcePath = file.Path.Replace('/', Path.DirectorySeparatorChar),
+                            TargetPath = Path.Combine(dir, file.Stem.Replace('/', Path.DirectorySeparatorChar))
```

Comment:
I think it might break in case of absolute, incorrect paths - two cases that comes to my mind are:

`/Windows/` -->You will probably end up witch `\Windows\file.txt` path
// --> you will end up with `\`